### DISCUSSION TITLE
Form's data model not initialized with default values

### DISCUSSION THREAD

**Discussion by rafaeltuelho** [vazco/uniforms#966] [Q&A] [answered]
I noticed that the model remains empty after Autoform renders. I would expect the model to start with field default values.
The model only updates at the field's `onChange`. Is there a way to "force" Uniforms model initialize with fields' default values?

**Comment by radekmie** [comment, marked as answer]
Hi @rafaeltuelho! All default (initial) values will be copied to the model if and only if the field is required. [Here](https://uniforms.tools/playground#?N4IgDgTgpgzlAuIBcICCATd0YwAQAoApAZQHkA5YgYwAsoBbAQwEoQAacCAezBmVEYBXeFxiMAblGQAzRgBs4HISLGSAIlDmMAnsgCMABgMd0ASzEAjOVHQz5ikFoubk8CIKgcwWqlBpc5dCgIOwVPEGhGdFIAOzldJFkwjhhaBkZkEHx8ZlwAXgA-XGAAHRjcXCouGJh4XEYAK3F83BioAHdcVCb8YHq5OQBRCG4IGCRcNw82XEE4DVlBOXhxyfdPXABrKG12rgh0VYBtEpBBGNNpffoYU4BdXABfZgBuMoqqmrrUuiYW0vKFUmpng1gmAHIMFhYDBwWx3kD4NowFAIVwLA0oFR4HCERVIDxgvBTLAJgCgUDpCTAqgyZNkajcODahBTDEAOZw3BBRbLCGocFPeGAilUzToABCdKRKIhLLZnJmPKEfKZEsFj2FFM1eNw0AAjoJTNB0BMjuCxTTwXctU83jEEdJztjTNVKpF4FAAGryUzoRgiCD4H7pXLkoGfWq4cS-_2BlqNcQAOiq9DApmswbSTFeZV10Hgggg5Xw9C4QTkuUKxV1FRjcj9Af2pfLmlzDpFQMuBHrjcDSeCozwADJh9HY02IAORvsYEnrBz4DQw7WgQWi-U-kF4IwM6te3H9tOh3bV49dY97RVzx2PtUoyGmD6G4eIC0qB7vRPA1nfix7Qi67Fq0HS4CQFDUH-Eqsug7JQL-6QzI-jDPn2-zts8OTsCAMD-O0ACScRslAwyjKEDiMDA2gxFQsTEIIFj0CC5HhJR1G0TEqFxlIiT2FAmogEuDA8WcXzaNYtiPEAA) is an example playground that presents this logic. There's no way to force it otherwise.

Do you have a case where it's actually needed? Usually, schemas provide some sort of "cleaning" function that does exactly that.

**Reply by rafaeltuelho** [reply]
Hi @radekmie! I see what you explained in the playground, but I can't see it working in that way in my app.
I can see the default value rendered inside the Input Field, but not reflected in the form's model.
I'm referring to the model using the `formRef`:

```js
<AutoForm
  ref={ref => (this.formRef = ref)}
  placeholder={true}
  schema={this.state.formBridgeSchema} 
  onChangeModel={model => this.updateState(model)} 
  onSubmit={this.onFormSubmit} >
</AutoForm>
```

and even setting my field/prop as required and defaultValue I'm getting the form model empty right after it gest rendered:

```js
  componentDidUpdate(prevProps, prevState, snapshot) {
    console.debug('DroolsDynamicForm ->>> componentDidUpdate...');
    console.debug('this.formRef', this.formRef.getModel());
   // model empty here!
  }

  componentDidMount() {
    console.debug('DroolsDynamicForm ->>> componentDidMount...');
    console.debug('this.formRef', this.formRef.getModel());
   // model empty here!
  }
```

**Reply by radekmie** [reply]
As I said, it'll appear in the model **only** if it's marked as required. Otherwise, it's going to be rendered, but not present in the model.
